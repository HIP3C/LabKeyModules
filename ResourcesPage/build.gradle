String moduleName = "${project.name}"
String moduleBuildDir = "${project.parent.labKeyModulesDir}/${moduleName}"

if(project.parent.reactModules.contains(moduleName)){
    npm_run_build {
        // input files listed are monitored for changes
        // and only rebuilt when needed
        // Note: Cannot use '*' for string matching, so
        // may need to specify different files
        inputs.files fileTree("src")
        inputs.files fileTree("webpack")
        inputs.file "package.json"
        inputs.file "package-lock.json"
        inputs.file "module.properties"
        inputs.file "postcss.config.js"
        inputs.file "tsconfig.json"
        inputs.file "declarations.d.ts"
        
        outputs.dir "resources" // set in where ??
    }

    task copyImageResources {
        dependsOn npm_run_build
        String imgDir = "${moduleBuildDir}/resources/web/ResourcesPage/images"
        mkdir imgDir
        copy {
            from "./images"
            include "*"
            into imgDir
        }
    }

    task zipNpmModule(type: Zip, dependsOn: 'copyImageResources') {
        from "${moduleBuildDir}/resources"
        archiveFileName = "${moduleName}.module"
        destinationDirectory = file("${project.parent.externalModulesDir}")
    }

    deploy {
        dependsOn zipNpmModule
    }

}else if(moduleName == 'extraWebApp'){
    deploy.doLast {
        mkdir "${project.parent.deployDir}/extraWebapp"
        copy {
            from "."
            include "*"
            into "${project.parent.deployDir}/extraWebapp"
        }
    }
    
}else{
    task zipModule(type: Zip)  {
        from "."
        archiveFileName = "${moduleName}.module"
        destinationDirectory = file("${project.parent.externalModulesDir}")
    }
    
    deploy {
        dependsOn zipModule
    }
}