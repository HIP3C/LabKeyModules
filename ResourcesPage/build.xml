<?xml version='1.0' encoding='UTF-8'?>

<project name='LabKeyModules' default='deploy' basedir='.'>

    <property environment='env'/>

    <target name='init'>
        <property name='modules.dir' value='${env.MODULES_DIR}'/>
        <property name='moduleDir' location='${basedir}'/>
        <basename property='module.name' file='${moduleDir}'/>
        <property name='module.src.dir' value='${basedir}'/>
        <property name='build.modules.dir' location='${basedir}/..'/>
    </target>

    <target name='module.properties' depends='init'>
        <ant antfile='../build_common.xml' target='ensure_enlistment_id' />
        <property file='enlistment.properties'/>

        <mkdir dir='${module.src.dir}/config'/>
        <tstamp>
            <format property='buildtime' pattern='MMMM d yyyy, h:mm aa'/>
        </tstamp>
        <copy tofile='${module.src.dir}/config/module.xml' file='${build.modules.dir}/module.template.xml' overwrite='true'>
            <filterset begintoken='@@' endtoken='@@'>
                <filtersfile file='${basedir}/module.properties'/>
                <filter token='BuildOS' value='${os.name}'/>
                <filter token='BuildPath' value='${build.modules.dir}/${module.name}.module'/>
                <filter token='BuildTime' value='${buildtime}'/>
                <filter token='BuildType' value='${BuildType}'/>
                <filter token='BuildUser' value='${user.name}'/>
                <filter token='EnlistmentId' value='${enlistment.id}'/>
                <filter token='ModuleDependencies' value=''/>
                <filter token='RequiredServerVersion' value='${RequiredServerVersion}'/>
                <filter token='Revision' value='Not built from a Subversion source tree'/>
                <filter token='SourcePath' value='${moduleDir}'/>
                <filter token='URL' value='Not built from a Subversion source tree'/>
            </filterset>
        </copy>
    </target>

    <target name='build' depends='module.properties'>
        <zip destfile='${build.modules.dir}/${module.name}.module'>
            <zipfileset dir='.'/>
        </zip>

        <available property='modulesDirExists' file='${modules.dir}'/>
    </target>

    <target name='deploy' depends='before_development_build,build' if='modulesDirExists'>
        <echo message='Deploying ${module.name}.module to ${modules.dir}'/>
        <copy file='${build.modules.dir}/${module.name}.module' overwrite='true' tofile='${modules.dir}/${module.name}.module'/>
        <delete file='${build.modules.dir}/${module.name}.module'/>
    </target>

    <target name='before_local_build'>
	<property name='BuildType' value='Local'/>
    </target>

    <target name='local' depends='before_local_build,deploy'></target>
	
    <target name='before_development_build'>
        <property name='BuildType' value='Development'/>
    </target>

    <target name='dev' depends='before_development_build,deploy'></target>

    <target name='before_production_build'>
        <property name='BuildType' value='Production'/>
    </target>

    <target name='prod' depends='before_production_build,deploy'></target>

</project>

