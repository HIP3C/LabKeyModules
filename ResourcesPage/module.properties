Description: Webparts for the Resources Page
Label: Resources Page
License: Apache 2.0
LicenseURL: https://www.apache.org/licenses/LICENSE-2.0
ModuleClass: org.labkey.api.module.SimpleModule
ModuleDependencies: ISCore
Name: ResourcesPage
Author: ImmuneSpace Development Team
Maintainer: ImmuneSpace Development Team
Organization: ImmuneSpace Development Team
OrganizationURL: https://www.immunespace.org
RequiredServerVersion: 14.1
SchemaVersion: 0.01
