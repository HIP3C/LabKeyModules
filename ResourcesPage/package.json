{
  "name": "resourcespage",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "setup": "npm install",
    "build": "npm run build-prod",
    "start": "cross-env NODE_ENV=development LK_MODULE=ResourcesPage webpack serve --config ../webpack/dev.config.js",
    "build-prod": "npm run clean && cross-env NODE_ENV=production LK_MODULE=ResourcesPage webpack --config ../webpack/prod.config.js",
    "clean": "rimraf resources/web/ResourcesPage/gen && rimraf resources/views",
    "test": "cross-env NODE_ENV=test jest"
  },
  "dependencies": {
    "bootstrap": "^3.4.1",
    "react-iframe": "^1.8.0",
    "regenerator-runtime": "^0.13.3"
  },
  "devDependencies": {
    "@types/jest": "27.0.2",
    "cross-env": "5.2.1",
    "jest": "27.2.5",
    "prettier": "2.3.1",
    "pretty": "2.0.0",
    "rimraf": "3.0.2",
    "ts-jest": "27.0.5",
    "webpack": "5.22.0",
    "webpack-cli": "4.9.0",
    "webpack-dev-server": "^4.7.3"
  },
  "jest": {
    "globals": {
      "LABKEY": {}
    },
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.setup.ts"
    ],
    "moduleNameMapper": {
      "\\.(scss)$": "<rootDir>/test/styleMock.ts"
    },
    "testRegex": "(\\.(test|spec))\\.(ts|tsx)$",
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    }
  },
  "typings": "./src/client/typings/main.d.ts"
}
