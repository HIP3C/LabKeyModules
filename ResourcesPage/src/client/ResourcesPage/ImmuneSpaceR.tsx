import * as React from "react";
import Iframe from 'react-iframe';

export const ImmuneSpaceR = () => {
    return (
        <div id="ImmuneSpaceR" style={{alignItems: "center", display: "flex", justifyContent: "center"}}>
            <Iframe
                url="https://rglab.github.io/ImmuneSpaceR/"
                width="90%"
                height="450px"
            />
        </div>
    )
}