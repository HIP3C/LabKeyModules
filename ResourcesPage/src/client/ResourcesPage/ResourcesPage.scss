@import '../theme/_ResourcesPage.scss';
  
.hr-infoCol {
    width: 70%;
    padding: 10px
}

.hr-imgCol {
    width: 15%
}

.hr-staticImg {
    margin: auto;
    width: 80%; //This is percent of col
    height: auto;
    border: 2px solid black;
    padding: 5px
}

