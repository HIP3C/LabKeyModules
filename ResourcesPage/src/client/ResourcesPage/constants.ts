export const TAB_REPORTS = "Reports";
export const TAB_STUDYSTATS = "StudyStats";
export const TAB_MOSTACCESSED = "MostAccessed";
export const TAB_MOSTCITED = "MostCited";
export const TAB_SIMILARSTUDIES = "SimilarStudies";
export const TAB_TOOLS = "Tools";
export const TAB_IMMUNESPACER = "ImmuneSpaceR";

export const tabInfo = [
  {
    id: "reports",
    tag: TAB_REPORTS,
    text: "Highlighted Reports",
  },
  {
    id: "study-stats",
    tag: TAB_STUDYSTATS,
    text: "Study Statistics",
    subMenu: [
      {
        id: "most-accessed",
        tag: TAB_MOSTACCESSED,
        text: "Most Accessed",
      },
      {
        id: "most-cited",
        tag: TAB_MOSTCITED,
        text: "Most Cited",
      },
      {
        id: "similar-studies",
        tag: TAB_SIMILARSTUDIES,
        text: "Similar Studies",
      },
    ],
  },
  {
    id: "tools",
    tag: TAB_TOOLS,
    text: "HIPC Tools",
  },
  {
    id: "immunespacer",
    tag: TAB_IMMUNESPACER,
    text: "ImmuneSpaceR",
  },
];

export const TOCITEMS = [
  { name: "Most Accessed", link: "#most-accessed" },
  { name: "Most Cited", link: "#most-cited" },
  { name: "Similar Studies", link: "#similar-studies" },
];
