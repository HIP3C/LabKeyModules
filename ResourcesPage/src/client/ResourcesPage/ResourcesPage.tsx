/*  ----------------
       Imports
------------------ */

// Libraries
import * as React from 'react';
import * as LABKEY from '@labkey/api';
import {MenuItem, Nav, Navbar, NavDropdown, NavItem, Tab, TabPane} from 'react-bootstrap';
import 'regenerator-runtime/runtime';

// Components
import {ImmuneSpaceR} from "./ImmuneSpaceR";
import {Tools} from "./Tools";
import {Reports} from "./Reports";

// Helpers
import {transformLogData, transformCiteData, transformSdyMetaData} from "./StudyStatsTransformationFunctions"
import {fetchApiData} from "./FetchApiData"

// Stylings
import './ResourcesPage.scss';

// Typings
import {BarPlotDatum} from "./components/mostCitedBarPlot"
import {ScatterPlotDatum, ScatterPlotDataRange} from "./components/similarStudyScatterPlot"
import {
    TAB_REPORTS,
    TAB_STUDYSTATS,
    TAB_MOSTACCESSED,
    TAB_MOSTCITED,
    TAB_SIMILARSTUDIES,
    TAB_TOOLS,
    TAB_IMMUNESPACER,
    tabInfo
} from "./constants";
import { MostAccessed } from './MostAccessed';
import { MostCited } from './MostCited';
import { SimilarStudies } from './SimilarStudies'

/*  ----------------
   StudyStats Data
------------------ */

const labkeyBaseUrl = LABKEY.ActionURL.getBaseURL()
const apiBase = labkeyBaseUrl + '_rapi/'

// RAW

const [pmData, setPmData] = React.useState({}); 
const [pmHasError, setPmErrors] = React.useState(false);

const [ssData, setSsData] = React.useState({}) 
const [ssHasError, setSsErrors] = React.useState(false)

const [maData, setMaData] = React.useState({}); 
const [maHasError, setMaErrors] = React.useState(false);

React.useEffect(() => {
    fetchApiData(apiBase, "pubmed_data", setPmData, setPmErrors);
    fetchApiData(apiBase, "study_data", setSsData, setSsErrors);
    fetchApiData(apiBase, "log_data", setMaData, setMaErrors);
}, []);

// TRANSFORMED

const [transformedPmData, setTransformedPmData] = React.useState({
    byPubId: Array<BarPlotDatum>()
});
const [pmDataRange, setPmDataRange] = React.useState({
    byPubId: Array<number>()
});

const [transformedSsData, setTransformedSsData] = React.useState(
    Array<ScatterPlotDatum>()
)
const [ssDataRange, setSsDataRange] = React.useState<ScatterPlotDataRange>({x: [], y: []})

const [transformedMaData, setTransformedMaData] = React.useState({
    byStudy: Array<Object>(),
    byMonth: Array<Object>()
});

React.useEffect(() => {
    transformSdyMetaData(ssData, setTransformedSsData, setSsDataRange)
    transformLogData(maData, setTransformedMaData)
    transformCiteData(pmData, setTransformedPmData, setPmDataRange)
}, [ssData, maData, pmData]);

// Handling of errors in data retrieval or transformation?

/*  ----------------
      Main
------------------ */

const ResourcesPage: React.FC = () => {

    const generateChildId = React.useCallback((eventKey: any, type: any) => {
        return eventKey;
    }, []);

    const getNavbar = React.useCallback(() => {
        const items = tabInfo.map((tab, index) => {

            // Dropdown
            if(tab.subMenu && tab.subMenu.length > 0) {
                const menuItems = tab.subMenu.map((sub) => {
                    return (
                        <MenuItem eventKey={sub.tag} id={sub.tag} key={sub.tag}>{sub.text}</MenuItem>
                    )
                });

                return (<NavDropdown title={tab.text} id={tab.id} key={tab.tag}>{menuItems}</NavDropdown>)
            }

            // Non-dropdown
            return (
                <NavItem eventKey={tab.tag} key={tab.tag}>{tab.text}</NavItem>
            )
        })

        return (
            <Navbar>
                <Nav style={{marginLeft:'0'}}>
                    {items}
                </Nav>
            </Navbar>
        )
    }, []);

    const getTabContent = React.useCallback(() => {
        return (
            <Tab.Content>
                <TabPane eventKey={TAB_REPORTS}>
                    <Reports/>
                </TabPane>
                <TabPane eventKey={TAB_MOSTACCESSED}>
                    <MostAccessed
                        transformedMaData={transformedMaData}
                        labkeyBaseUrl={labkeyBaseUrl}
                    />
                </TabPane>
                <TabPane eventKey={TAB_MOSTCITED}>
                    <MostCited
                        transformedPmData={transformedPmData}
                        pmDataRange={pmDataRange}
                    />
                </TabPane>
                <TabPane eventKey={TAB_SIMILARSTUDIES}>
                    <SimilarStudies
                        transformedSsData={transformedSsData}
                        ssDataRange={ssDataRange}
                        labkeyBaseUrl={labkeyBaseUrl}
                    />
                </TabPane>
                <TabPane eventKey={TAB_TOOLS}>
                    <Tools/>
                </TabPane>
                <TabPane eventKey={TAB_IMMUNESPACER}>
                    <ImmuneSpaceR/>
                </TabPane>
            </Tab.Content>
        )
    }, [transformedPmData, transformedSsData, transformedMaData]);

    return(
        <Tab.Container defaultActiveKey={TAB_REPORTS} generateChildId={generateChildId}>
            <div>
                {getNavbar()}
                {getTabContent()}
            </div>
        </Tab.Container>
    )
}

// --------- EXPORT ------------
// There should be a single export: a component called "App"
export const App: React.FC = () => {

    const filterBanner = document.getElementById('filter-banner')
    filterBanner.style.display = 'none'

    // Must return a React Fragment
    return <ResourcesPage/>
}
