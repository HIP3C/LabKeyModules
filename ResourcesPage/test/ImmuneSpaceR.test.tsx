/**
 * @jest-environment jsdom
 */
import React from "react";
import { ImmuneSpaceR } from "../src/client/ResourcesPage/ImmuneSpaceR";
import renderer from "react-test-renderer";

describe("test ImmuneSpaceR renders correctly", () => {
  test("test ImmuneSpaceR renders correctly", () => {
    const component = renderer.create(<ImmuneSpaceR />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
