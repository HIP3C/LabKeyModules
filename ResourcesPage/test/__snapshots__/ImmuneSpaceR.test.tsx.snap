// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ImmuneSpaceR renders correctly test ImmuneSpaceR renders correctly 1`] = `
<main
  className="page-content"
>
  <section
    className="content-grid__immunespaceR"
  >
    <div
      className="immunespaceR-content grid-item"
    >
      <h1>
        ImmuneSpaceR
      </h1>
      <img
        alt="immunespaceR logo"
        className="immunespaceR-logo-img"
        id="immunespaceR-logo-mobile"
        src="/ResourcesPage/images/sticker3.svg"
      />
      <div
        className="immunespaceR-text"
      >
        <p>
          A thin wrapper around Rlabkey to access the ImmuneSpace database from R.
        </p>
        <p>
          This package simplifies access to the HIPC ImmuneSpace database for R programmers. It takes advantage of the standardization of the database to hide all the Rlabkey specific code away from the user. The study-specific datasets can be accessed via an object-oriented paradigm.
        </p>
      </div>
      <a
        className="arrow-link"
        href="https://github.com/RGLab/ImmuneSpaceR"
        title="Learn More"
      >
        <span
          className="tri-block__link-text"
        >
          Learn More
        </span>
      </a>
    </div>
    <div
      className="immunespaceR-logo grid-item"
    >
      <img
        alt="immunespaceR logo"
        className="immunespaceR-logo-img"
        id="immunespaceR-logo-desk"
        src="/ResourcesPage/images/sticker3.svg"
      />
    </div>
  </section>
</main>
`;
